mockito-core/src/main/java/org/mockito/internal/creation/bytebuddy/InlineDelegateByteBuddyMockMaker.java

```diff
@@ -554,7 +558,10 @@ public TypeMockability isTypeMockable(final Class<?> type) {
         return new TypeMockability() {
             @Override
             public boolean mockable() {
-                return INSTRUMENTATION.isModifiableClass(type) && !EXCLUDES.contains(type);
+                return INSTRUMENTATION.isModifiableClass(type)
+                        && !EXCLUDES.contains(type)
+                        && !(Modifier.isAbstract(type.getModifiers())
```

Comment:
Should we instead update `INSTRUMENTATION` to handle this? If that does not make sense, then I would vouch for creating a method that returns a boolean and we use it here. That makes it more readable and we can use it down below in the reason as well.

Reply:
That's not possible as the instrumentation comes from the JDK.

But I was thinking about moving some type related checks in our `TypeSupport`. Currently, it only has a check for `isSealed`.

Reply:
WDYT about introducing a `TypeSupport.INSTANCE.ismockable()` where we implement the logic? That way, the separation of concerns is on the class boundary.

Reply:
I'm not sure about moving that logic here at this time, because

1. The mockability depends on the mock maker
2. If we revisit this code, I think we may want to revisit `TypeMockability` as well, e.g. instead of returning string, I wonder if we could use a list of enum values instead.